--- origsrc/libxklavier-3.8/libxklavier/Makefile.am	2008-11-14 18:41:29.000000000 -0600
+++ src/libxklavier-3.8/libxklavier/Makefile.am	2008-12-11 14:29:57.362375000 -0600
@@ -45,7 +45,8 @@
 	xklavier_xmm.c xklavier_xmm_opts.c xklavier_evt_xmm.c xklavier_config_xmm.c \
 	xklavier_util.c xklavier_props.c xklavier_dump.c xkl_engine_marshal.c \
 	$(noinst_HEADERS) $(xklavierinc_HEADERS)
-libxklavier_la_LDFLAGS = -version-info @VERSION_INFO@ \
+libxklavier_la_LDFLAGS = -version-info @VERSION_INFO@ -no-undefined
+libxklavier_la_LIBADD = \
  $(XML_LIBS) $(GLIB_LIBS) $(XINPUT_LIBS) \
  $(LIBXKBFILE_PRESENT_LDFLAGS) \
  $(X_LIBS) $(LIBICONV) 
--- origsrc/libxklavier-3.8/tests/Makefile.am	2007-05-31 16:06:45.000000000 -0500
+++ src/libxklavier-3.8/tests/Makefile.am	2008-12-11 14:30:10.378000000 -0600
@@ -6,5 +6,5 @@
 
 AM_CFLAGS=-Wall -Werror -I$(includedir) $(X_CFLAGS) -I$(top_srcdir) $(GLIB_CFLAGS)
 
-AM_LDFLAGS=$(top_builddir)/libxklavier/libxklavier.la $(X_LIBS) -lX11 $(GLIB_LIBS)
+LDADD=$(top_builddir)/libxklavier/libxklavier.la $(X_LIBS) -lX11 $(GLIB_LIBS)
 
