--- origsrc/libxklavier-3.4/libxklavier/Makefile.am	2007-10-05 17:43:24.000000000 -0500
+++ src/libxklavier-3.4/libxklavier/Makefile.am	2008-01-30 23:47:54.953125000 -0600
@@ -44,9 +44,10 @@
 	xklavier_xmm.c xklavier_xmm_opts.c xklavier_evt_xmm.c xklavier_config_xmm.c \
 	xklavier_util.c xklavier_props.c xklavier_dump.c xkl_engine_marshal.c \
 	$(noinst_HEADERS) $(xklavierinc_HEADERS)
-libxklavier_la_LDFLAGS = -version-info @VERSION_INFO@ \
+libxklavier_la_LDFLAGS = -version-info @VERSION_INFO@ -no-undefined
+libxklavier_la_LIBADD = \
  $(XML_LIBS) $(GLIB_LIBS) \
  $(LIBXKBFILE_PRESENT_LDFLAGS) \
- $(X_LIBS) $(LIBICONV) 
+ $(X_LIBS) -lX11 $(LIBICONV) 
 
 
--- origsrc/libxklavier-3.4/tests/Makefile.am	2007-05-31 16:06:45.000000000 -0500
+++ src/libxklavier-3.4/tests/Makefile.am	2008-01-30 23:47:54.968750000 -0600
@@ -6,5 +6,5 @@
 
 AM_CFLAGS=-Wall -Werror -I$(includedir) $(X_CFLAGS) -I$(top_srcdir) $(GLIB_CFLAGS)
 
-AM_LDFLAGS=$(top_builddir)/libxklavier/libxklavier.la $(X_LIBS) -lX11 $(GLIB_LIBS)
+LDADD=$(top_builddir)/libxklavier/libxklavier.la $(X_LIBS) -lX11 $(GLIB_LIBS)
 
